common/src/test/java/com/revenuecat/purchases/common/networking/ETagManagerTest.kt

```diff
@@ -50,24 +56,66 @@ class ETagManagerTest {
         val path = "/v1/subscribers/appUserID"
         mockCachedHTTPResult(expectedETag = null, path = path)
 
-        val requestWithETagHeader = underTest.getETagHeader(path)
-        val eTagHeader = requestWithETagHeader[HTTPRequest.ETAG_HEADER_NAME]
+        val eTagHeaders = underTest.getETagHeaders(path)
+        val eTagHeader = eTagHeaders[HTTPRequest.ETAG_HEADER_NAME]
         assertThat(eTagHeader).isNotNull
         assertThat(eTagHeader).isBlank
     }
 
+    @Test
+    fun `ETag Last refresh time header is null if there is no ETag saved for that request`() {
+        val path = "/v1/subscribers/appUserID"
+        mockCachedHTTPResult(expectedETag = null, path = path)
+
+        val eTagHeaders = underTest.getETagHeaders(path)
+        val lastRefreshTimeHeader = eTagHeaders[HTTPRequest.ETAG_LAST_REFRESH_NAME]
```

Comment:
This will return null both if the value is null or if the key is not present. I asked in another comment, but do we want to even have the header if the value is null?

Reply:
As mentioned in the other comment, we will filter the header out in the HTTPClient so it won't actually be sent.